examples/cosmos/lightclient/Blockchain.qnt

```diff
@@ -0,0 +1,168 @@
+
+    import typedefs.* from "./typedefs"
+
+    // a set of all nodes that can act as validators (correct and faulty)
+    const AllNodes: Set[NODE]
+    // the maximal height that can be ever reached (modelling artifact)
+    const ULTIMATE_HEIGHT: int
+    // the period within which the validators are trusted
+    const TRUSTING_PERIOD: int
+
+    // possible heights
+    pure def Heights = 1.to(ULTIMATE_HEIGHT)
+
+    // A commit is just a set of nodes who have committed the block
+    pure def Commits = powerset(AllNodes)
+
+    // The set of all block headers that can be on the blockchain.
+    // This is a simplified version of the Block data structure in the
+    // actual implementation.
+    pure def BlockHeaders = 
+        tuples(Heights, Int, Commits, powerset(AllNodes), powerset(AllNodes)).map((
+            // the block height
+            height,
+            // the block timestamp in some integer units
+            time,
+            // the nodes who have voted on the previous block, the set itself
+            // instead of a hash in the implementation, only the hashes of
+            // V and NextV are stored in a block, as V and NextV are stored
+            // in the application state
+            lastCommit,
+            // the validators of this bloc. We store the validators instead of the hash
+            VS,
+            // the validators of the next block.
+            // We store the next validators instead of the hash.
+            NextVS) => { 
+                height: height, 
+                time: time,
+                lastCommit: lastCommit,
+                VS: VS,
+                NextVS: NextVS
+            }
+        )
+
+    // A signed header is just a header together with a set of commits 
+    pure def LightBlocks = tuples(BlockHeaders, Commits).map((header, commits) => {header: header, Commits: commits})
+
+
+    // VARIABLES
+    // Current global time in integer units as perceived by the reference chain
+    var refClock: int
+    // A function of heights to BlockHeaders,
+    // which gives us a bird view of the blockchain.
+    var blockchain: Blockchain
+    // A set of faulty nodes, which can act as validators.
+    // We assume that the set of faulty processes is non-decreasing.
+    // If a process has recovered, it should connect using a different id.
+    var Faulty: Set[NODE]
+
+
+    // The set of all correct nodes in a state
+    def Corr = AllNodes.exclude(Faulty)
+
+
+    // (****************************** BLOCKCHAIN ************************************)
+
+    // the header is still within the trusting period
+    def InTrustingPeriod(header: BlockHeader): bool = refClock < header.time + TRUSTING_PERIOD
+    
+    //  Given a function pVotingPower \in D -> Powers for some D \subseteq AllNodes
+    //  and pNodes \subseteq D, test whether the set pNodes \subseteq AllNodes has
+    //  more than 2/3 of voting power among the nodes in D.
```

Comment:
I don't think we need this scary math notation in the comments anymore

Reply:
Well, this _is_ still LateX syntax. Do you want to just delete the comment? Writing Quint in the comments doesn't really make sense.